machine/v1/types_aws.go

```diff
@@ -20,6 +20,7 @@ type AWSResourceReference struct {
 	ARN *string `json:"arn,omitempty"`
 	// Filters is a set of filters used to identify a resource.
 	// +optional
+	// +listType=atomic
```

Comment:
I'm not familiar with this type of list, what does it do?

Reply:
Atomic lists are updated as a unit. This is actually the default behaviour when specified, hence, in most cases, the correct thing to do for backward compat is to set the default explicitly.

The alternatives are map and set, maps I believe you've seen before, and sets are problematic, so we avoid them